### ISSUE TITLE
Provide Stochastic Gradient Descent in some ML package

### ISSUE THREAD

**Issue by buddha314** [chapel-lang/chapel#6847]
Towards a suite of machine learning functionality, here is a fundamental routine for solving systems.

A' la [SciPy's implementation](http://scikit-learn.org/stable/modules/sgd.html) of [Stochastic Gradient Descent](https://en.wikipedia.org/wiki/Stochastic_gradient_descent)

**Comment by buddha314** [comment]
Seems like only yesterday I wrote this ticket...   Anyway, this would be a PERFECT thing to add to [NumSuch](https://github.com/buddha314/numsuch) and we have some of the underlying tools installed.
